src/dsymbol/symbol.d

```diff
@@ -375,10 +375,16 @@ struct DSymbol
 	/**
 	 * Names of function arguments
 	 */
+	// TODO: remove since we have function arguments
 	UnrolledList!(istring) argNames;
 
 	private uint _location;
 
+	/**
+	 * Function argument symbols
+	 */
+	DSymbol*[] functionArguments;
```

Comment:
Does it need to be an array of pointers?

Reply:
I guess so since DSymbol isn't copyable unless we change that.
`/**
 * Autocompletion symbol
 */
struct DSymbol
{
	// Copying is disabled
	@disable this();
	@disable this(this);

	`

Reply:
usually all the DSymbol references are done by pointers, but it might be effective to use them inline in arrays (e.g. move them into the array)

We already know the count of items that get put into array before putting them there anyway, so we can already preallocate all the space and directly construct them in-place.